recipes/libxml2/all/test_package/CMakeLists.txt

```diff
@@ -1,17 +1,7 @@
 cmake_minimum_required(VERSION 3.1)
 project(test_package C)
 
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup(TARGETS)
-
-find_package(LibXml2 REQUIRED)
-message("LIBXML2_FOUND: ${LIBXML2_FOUND}")
-message("LIBXML2_INCLUDE_DIR: ${LIBXML2_INCLUDE_DIR}")
-message("LIBXML2_INCLUDE_DIRS: ${LIBXML2_INCLUDE_DIRS}")
-message("LIBXML2_LIBRARIES: ${LIBXML2_LIBRARIES}")
-message("LIBXML2_LIBRARY: ${LIBXML2_LIBRARY}")
-message("LIBXML2_DEFINITIONS: ${LIBXML2_DEFINITIONS}")
-message("LIBXML2_VERSION_STRING: ${LIBXML2_VERSION_STRING}")
+find_package(libxml2 CONFIG REQUIRED)
```

Comment:
I think we should actually add a `test_module_package/` 🤔 

Since there's an official https://cmake.org/cmake/help/latest/module/FindLibXml2.html one we need to make sure it not broken 

and we should test both 🤔 

Does that make sense? leave the old CMakeLists.txt and duplicate the rest?

Reply:
Yes, it's what I did in freetype recipe, it can be used as a source of inspiration to test module & config file (as well as CMake variables in module file): https://github.com/conan-io/conan-center-index/tree/master/recipes/freetype

Reply:
@SpaceIm re the libjpeg-turbo test_v1_package_module, why did you do this:
```
add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package_module
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package_module)
```

Doesn't that import the v2 cmake with the find_package call?
That doesn't test the v1 import mechanism?

Reply:
@prince-chrismc @SpaceIm should I ever be testing these old cmake variables, like LIBXML2_LIBRARIES and LIBXML2_FOUND ?
These still work with the new (soon to be pushed) test_v1_package_module, but nothing else.

Reply:
Should I be testing LIBXML2_INCLUDE_DIR and other old cmake variables in test_v1_package_module?
They work, but only if they are in the CMakeLists.txt in test_package_module.
If I print out the messages after `add_subdirectory()` (in test_v1_package_module), nothing is printed - cmake scoping?

Reply:
Definitely a scope issue. It's not setting parent by default.

I think the example space shared is a good way. Put all the code in one and add sub dir from the other.


I would call it "test cmake module package" so it's more clear but that a nit

Test package module implies the opposite of what it's doing in English

Reply:
Renamed

Reply:
I've re-added the LIBXML2_INCLUDE_DIR to the test_cmake_module_package test,
however,
it does NOT work correctly for test_cmake_module_package,
it only works for test_v1_cmake_module_package.

I'm wondering if we should test these at all?

Reply:
Sure it should be tested. If it fails it means that there is an issue in libxml2 recipe.

Reply:
@SpaceIm ok, well I am not sure if these should work at all.  Yes for old v1 stuff, but from other conversations I believe the goal is to move away from old cmake variables and use modern targets etc.
So I guess this recipe is fine for now?